### PR TITLE
Text reverse index

### PR DESCRIPTION
**Pull request by Jeanselme**
Creates a first version of text reverse indexing.
This one has to be improved in order to have a better search of the maximum.

### CODE REVIEW

File: ReverseIndexManip/AbstractReverseIndexManip.py [hunk 1]
```diff
@@ -11,42 +11,99 @@ class abstractReverseIndexManip:
...
+		"""
+		resKey = ''
+		resDocuments = []
+		resValues = []
+		for index, i in zip(self.indexes, range(len(self.indexes))):
+			# There is a key which is not null due to while condition
+			if index.currentValue != None:
+				key, value = index.currentValue[0], index.currentValue[1]
+				print(key, value)
+				if resKey == '' or self.compare(resKey,key) :
```
Comment (clinm) [inline ReverseIndexManip/AbstractReverseIndexManip.py]: First time I see a space before '**:**'. mistype ?

File: ReverseIndexManip/AbstractReverseIndexManip.py [hunk 2]
```diff
@@ -11,42 +11,99 @@ class abstractReverseIndexManip:
...
+	def selectKey(self):
+		"""
+		Selects the first key to put and returns docs and values
+		Private
+		"""
+		resKey = ''
+		resDocuments = []
+		resValues = []
+		for index, i in zip(self.indexes, range(len(self.indexes))):
+			# There is a key which is not null due to while condition
```
Comment (clinm) [inline ReverseIndexManip/AbstractReverseIndexManip.py]: Are you referring to the **while** in the create method ?

Comment (Jeanselme) [reply ReverseIndexManip/AbstractReverseIndexManip.py]: Yes this function is a private one which has to be called only if a key is not null, however I will add a comment

File: ReverseIndexManip/AbstractReverseIndexManip.py (hunk 1 above)
Comment (Jeanselme) [reply ReverseIndexManip/AbstractReverseIndexManip.py]: My mistake

**Pull request merged**